src/GitVersion.Core/VersionCalculation/VersionSearchStrategies/TrunkBasedVersionStrategy.cs

```diff
@@ -0,0 +1,335 @@
+using System.ComponentModel;
+using GitVersion.Common;
+using GitVersion.Configuration;
+using GitVersion.Core;
+using GitVersion.Extensions;
+using GitVersion.VersionCalculation.TrunkBased;
+using GitVersion.VersionCalculation.TrunkBased.NonTrunk;
+using GitVersion.VersionCalculation.TrunkBased.Trunk;
+
+namespace GitVersion.VersionCalculation;
+
+internal sealed class TrunkBasedVersionStrategy : VersionStrategyBase
+{
+    private volatile int IterationCounter;
+
+    private ITaggedSemanticVersionRepository TaggedSemanticVersionRepository { get; }
```

Comment:
please use the `private readonly ` fields instead of the private properties, as we do everywhere in the code

Reply:
Erl.